airflow-core/src/airflow/api/common/trigger_dag.py

```diff
@@ -85,10 +84,11 @@ def _trigger_dag(
                     f"[{min_dag_start_date.isoformat()}] from DAG's default_args"
                 )
         coerced_logical_date = timezone.coerce_datetime(logical_date)
-        data_interval = dag.timetable.infer_manual_data_interval(run_after=run_after)
+        data_interval = dag.timetable.infer_manual_data_interval(run_after=coerced_logical_date)
```

Comment:
This does not look right. The method takes a `run_after`, passing in something else is just wrong.